src/core/qgsmapsettings.h

```diff
@@ -366,6 +366,20 @@ class CORE_EXPORT QgsMapSettings
      */
     const QgsLabelingEngineSettings &labelingEngineSettings() const { return mLabelingEngineSettings; }
 
+    /**
+     * Returns maximum rendering time for layer rendering in preview jobs.
+     * \since QGIS 3.0
+     * \note not available in Python bindings
+     */
```

Comment:
I'm wondering if this really needs to be configurable. If we just moved the PREVIEW_JOB_DELAY_MS int to qgis.h then we could skip all the changes to map settings and QgsDataProvider::renderInPreview could drop the max render time argument too.